xbmc/guilib/GUIAudioManager.cpp

```diff
@@ -222,30 +226,23 @@ void CGUIAudioManager::UnLoad()
 
 std::string GetSoundSkinPath()
 {
-  const std::string id = CSettings::Get().GetString(CSettings::SETTING_LOOKANDFEEL_SOUNDSKIN);
-  if (id == "OFF")
+  auto setting = static_cast<CSettingString*>(CSettings::Get().GetSetting(CSettings::SETTING_LOOKANDFEEL_SOUNDSKIN));
+  auto value = setting->GetValue();
+  if (value.empty())
     return "";
 
-  if (id == "SKINDEFAULT")
-    return URIUtils::AddFileToFolder(g_SkinInfo->Path(), "sounds");
-
   ADDON::AddonPtr addon;
-  if (ADDON::CAddonMgr::Get().GetAddon(id, addon, ADDON::ADDON_RESOURCE_UISOUNDS))
-    return URIUtils::AddFileToFolder("resource://", id);
-
-  //Check special directories
-  std::string path = URIUtils::AddFileToFolder("special://home/sounds", id);
-  if (XFILE::CDirectory::Exists(path))
-    return path;
+  if (ADDON::CAddonMgr::Get().GetAddon(value, addon, ADDON::ADDON_RESOURCE_UISOUNDS))
+    return URIUtils::AddFileToFolder("resource://", value);
```

Comment:
You could reverse the logic and check if the addon doesn't exist and if that's the case, write the log message, reset the setting and update `value`. And then you could
```
return URIUtils::AddFileToFolder("resource://", value);
```
That saves you the duplication of the `URIUtils::AddFileToFolder()` logic.